static/app/views/settings/components/teamSelect.tsx

```diff
@@ -92,6 +92,32 @@ function TeamSelect({
   const {teams, onSearch, fetching} = useTeams();
   const {orgRoleList, teamRoleList} = organization;
 
+  const slugsToFilter: string[] =
+    selectedTeams?.map(tm => tm.slug) || selectedTeamRoles?.map(tm => tm.teamSlug) || [];
+
+  // determine if adding a team changes the minimum team role
+  const getTopOrgRole = (): string | undefined => {
+    const teamsWithOrgRole = teams.filter(
+      team => slugsToFilter.includes(team.slug) && team.orgRole !== null
+    );
+    // get teams with org roles, if any
+    const allOrgRoles =
+      teamsWithOrgRole.length > 0
+        ? teamsWithOrgRole.map(team => team.orgRole ?? undefined)
+        : [];
+    // sort them and to get the highest priority role
+    // highest prio role may change minimum team role
+    allOrgRoles.push(selectedOrgRole);
+    allOrgRoles.sort((a, b) =>
+      orgRoleList.findIndex(r => r.id === a) < orgRoleList.findIndex(r => r.id === b)
```

Comment:
🤔 hmmm....
this feels very smelly.... index is not necessarily priority right?

Reply:
in this case, index is priority because the `orgRoleList` has them in order